### PR TITLE
Revised syntactically-correct.cell

### PR DESCRIPTION
**Pull request by falkecarlsen**
- Now uses floats as numbers. This is dependent on #17.
- Defines a function which takes a `neighbourhood` and a `state` as parameters. This is dependent on #23.
- Fixed missing curly-brackets around if-statement body.

Does not utilise all grammar facilities.

### CODE REVIEW

File: parser/syntax_testing/syntactically-correct.cell [hunk 1]
```diff
@@ -14,27 +14,21 @@ neighbourhood foobar {
     (-2,-2), (-1,-2), (0,-2), (1,-2), (2,-2)
 }
 
-/* This state has a
```
Comment (MathiasMehl) [inline parser/syntax_testing/syntactically-correct.cell]: Why did you remove the multiline comment, as this file should test all grammar facilities

File: parser/syntax_testing/syntactically-correct.cell [hunk 2]
```diff
@@ -14,27 +14,21 @@ neighbourhood foobar {
     (-2,-2), (-1,-2), (0,-2), (1,-2), (2,-2)
 }
 
-/* This state has a 
-long, multi-line
-comment */
-state foo (255, 0, 0) {
-    let a = (not (2 - 3) / (4)) is 2;
```
Comment (MathiasMehl) [inline parser/syntax_testing/syntactically-correct.cell]: Add this line 21 to "bar" state, as it tests grammar not currently tested

**Review by MathiasMehl — CHANGES_REQUESTED** [review]
Make example for-loop, as this is not currently tested in this file.

File: parser/syntax_testing/syntactically-correct.cell [hunk 3]
```diff
@@ -14,27 +14,21 @@ neighbourhood foobar {
...
     }
     // single stmt after if-stmt
-    if (count(foobar, foo) is 2) become foo;   
+    if (func(foobar, foo) == 2) {
+        become foo;
+    }
 }
+
+function func (neighbourhood foo, state bar) state {
+    if (func2(foobar)) {
```
Comment (NicEastvillage) [inline parser/syntax_testing/syntactically-correct.cell]: Er det meningen at `func2` bliver kaldt, når den ikke er defineret?

**Review by jenrik — DISMISSED** [review]
Missing examples of:
 - Most expressions
 - Multi state index, `#`
 - Multi state declaration
 - `elif`
 - Array literal
 - Maybe other things

**Comment by NicEastvillage** [comment]
Adding to @jenrik's list of missing features:

* for-loops
* continue-statement
* break-statement
* neighbourhood lookup, e.g. `moore[0] == Alive`

File: parser/syntax_testing/syntactically-correct.cell (hunk 3 above)
Comment (falkecarlsen) [reply parser/syntax_testing/syntactically-correct.cell]: Should not matter for simply syntax, but could be revised to a `func` call, implying recursivity or `count`, testing built-in function-syntax.

**Comment by falkecarlsen** [comment]
@NicEastvillage @jenrik, I agree with the list of missing syntax-examples, however, this PR aims to only correct the faulty version currently on `master` not complete an example program with every construct.

Please create an issue, which we can refer to and add to, as the grammar evolves.

File: parser/syntax_testing/syntactically-correct.cell (hunk 3 above)
Comment (falkecarlsen) [reply parser/syntax_testing/syntactically-correct.cell]: Revised to `func` for recursion.

File: parser/syntax_testing/syntactically-correct.cell (hunk 2 above)
Comment (falkecarlsen) [reply parser/syntax_testing/syntactically-correct.cell]: The state `foo` has been added back.

File: parser/syntax_testing/syntactically-correct.cell (hunk 1 above)
Comment (falkecarlsen) [reply parser/syntax_testing/syntactically-correct.cell]: The state `foo` has been added back.

**Comment by falkecarlsen** [comment]
@jenrik @MathiasMehl I've fixed the existing issues and the `syntactically-correct.cell` is now actually syntactical correct. Issue #25 collects the missing facilities. Please re-review.

**Review by MathiasMehl — APPROVED** [review]
